### DISCUSSION TITLE
How does Tabby parse/process the git repository codes into context of code search?

### DISCUSSION THREAD

**Discussion by ANYMS-A** [TabbyML/tabby#3272] [Q&A]
Hi, I recently tried to figure out that:  how Tabby process a git repo (to AST) and save the processed into the Database or VectorDB for context searching at code completion stage？  May I ask which part of the code contains the implementation of this feature?

I saw some CodeSearchSchema declaration in the `crates/tabby-common/src/index.rs`  and had a rough view about the Schema, but I'd like to know more detail about the whole pipeline of how a git repo of code is processed and stored.

 Thanks!

**Comment by Cfomodz** [comment]
```
Git Repo → Sync → Parse → Chunk → Embed → Tantivy Index
```
